### ISSUE TITLE
Convection's discrete LCL/LFC/EL argmax picks make every convection parameter non-differentiable

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#665]
Found by a re-audit of the July 2026 ECHAM+RRTMGP review against `dev` @ `f6d1bcd`. Parent: #614.

## What's wrong

Three discrete `argmax` picks select the levels that define the convective column, and they were untouched by #556:

- `tiedtke_nordeng.py:161` — `cloud_base_level = jnp.argmax(saturated_pressure)`
- `tiedtke_nordeng.py:283` — `lfc_sf = jnp.argmax(pos_above_cb)`
- `tiedtke_nordeng.py:292` — `el_sf = jnp.argmax(el_candidate_mask)`

The CAPE integral is then a hard `where((k >= lfc_sf) & (k < el_sf))` mask (`:295-306`). Because these indices step discontinuously, everything downstream of them is a step function of the ambient state.

## Evidence

Smooth RH sweep, 20-level tropical column, `t_sfc = 300 K`, `moisture_supply = 2e-5`:

```
rh=0.20 cb=12 CAPE= 2971.9 precip=3.68e-04 | d/dentrpen=-9.14e-01 d/dcprcon= 9.23e-01
rh=0.30 cb=13 CAPE= 4942.0 precip=8.50e-04 | d/dentrpen=-1.46e+00 d/dcprcon= 1.67e+00
rh=0.40 cb=14 CAPE= 7366.1 precip=1.81e-21 | d/dentrpen= 0.00e+00 d/dcprcon= 0.00e+00
rh=0.50 cb=15 CAPE= 9880.7 precip=1.26e-21 | d/dentrpen= 0.00e+00 d/dcprcon= 0.00e+00
rh=0.60 cb=16 CAPE=11789.5 precip=1.64e-06 | d/dentrpen=-7.29e-03 d/dcprcon= 1.21e-02
rh=0.85 cb=18 CAPE=15578.6 precip=6.79e-05 | d/dentrpen=-4.67e-02 d/dcprcon= 1.01e-01
```

Convective precipitation moves **non-monotonically over 17 orders of magnitude** under a smooth 0.1 change in ambient RH, purely because the cloud-base index steps 12→13→14→…→18. At two of nine samples every convection parameter gradient is exactly zero. Separately, `d(CAPE)/d(t_surf)` from AD is a staircase (31.9 → 43.0 → 45.3 → 58.5 → 58.3 over 0.25 K steps) that matches finite differences only piecewise — the LFC/EL indices are locked within each tread.

## Why it matters

No convection parameter — `entrpen`, `cprcon`, `tau`, `cmfctop`, `zdnoprc` — can be calibrated by gradient descent against any target whose optimisation trajectory moves a column across a cloud-base/LFC/EL boundary. That is every realistic trajectory. Of everything in the smoothing backlog this is the item that most undermines "differentiable convection", and it is the hardest.

## Fix

Replace the three argmax picks with sigmoid-weighted soft masks over the level index:

- **LFC/EL:** `w_k = sigmoid(buoyancy_k / w_b)`, cumulatively gated to form a soft window, so `CAPE = Σ w_k · buoy_k · dz_k`.
- **Cloud base:** either a softmax over `(surf_humid - parcel_qs)/w_q` weighting `pressure[k]` to give a *continuous* base pressure, or interpolate the base properties between the two bracketing levels.

New widths `smooth_lfc_b` (m/s²) and `smooth_lcl_q` (kg/kg) belong beside the existing `smooth_term_buoy` at `types.py:64-76`, annealable like the others.

Two dependencies worth noting: the companion clips/floors issue widens `smooth_term_buoy`, which removes a *second* source of the exact zeros above; and the cloud-base fix interacts with the cloud-base saturation adjustment issue, since `find_cloud_base`'s argmax is what leaves the parcel subsaturated there.

Also still hard, lower priority, same technique: the downdraft LFS argmax and wet-bulb `lax.cond` (`downdraft.py:169-170, 79`), which give `cmfdeps`/`entrdd` piecewise-constant gradients. `sundqvist._stratocumulus_zsat` (`sundqvist.py:311-321`) is the worked example of the softmax replacement already in the tree.

## Refs

Maintainability review B.1 (Tiedtke, "discrete LCL/LFC/EL picks"), B.2.2.
